llvm/tools/sycl-post-link/ModuleSplitter.h

```diff
@@ -36,19 +38,84 @@ using EntryPointVec = std::vector<const Function *>;
 
 struct EntryPointGroup {
   StringRef GroupId;
+
+public:
   EntryPointVec Functions;
+
+  EntryPointGroup(StringRef GroupId = "") : GroupId(GroupId) {}
+  EntryPointGroup(StringRef GroupId, EntryPointVec &&Functions)
+      : GroupId(GroupId), Functions(std::move(Functions)) {}
+
+  const StringRef getId() const { return GroupId; }
+  bool isEsimd() const;
+  bool isSycl() const;
+  void saveNames(std::vector<std::string> &Dest) const;
+  void rebuildFromNames(const std::vector<std::string> &Names, const Module &M);
+  void rebuild(const Module &M);
 };
 
 using EntryPointGroupVec = std::vector<EntryPointGroup>;
 
-struct ModuleDesc {
+enum class ESIMDStatus { SYCL_ONLY, ESIMD_ONLY, SYCL_AND_ESIMD };
+
+class ModuleDesc {
   std::unique_ptr<Module> M;
   EntryPointGroup EntryPoints;
 
+public:
+  struct Properties {
+    ESIMDStatus HasEsimd = ESIMDStatus::SYCL_AND_ESIMD;
+    bool SpecConstsMet = true;
+  };
+  std::string Name = "";
+  Properties Props;
+
+  ModuleDesc(std::unique_ptr<Module> &&M, EntryPointGroup &&EntryPoints)
+      : M(std::move(M)), EntryPoints(std::move(EntryPoints)) {
+    Name = this->EntryPoints.getId().str();
+  }
+
+  ModuleDesc(std::unique_ptr<Module> &&M, const std::vector<std::string> &Names)
+      : M(std::move(M)) {
+    rebuildEntryPoints(Names);
+  }
+
+  bool isESIMD() const { return Props.HasEsimd == ESIMDStatus::ESIMD_ONLY; }
+  bool isSYCL() const { return Props.HasEsimd == ESIMDStatus::SYCL_ONLY; }
```

Comment:
'HasEsimd == ESIMDStatus::SYCL_ONLY' is a bit awkward, wouldn't it be better to say something like:
'Props.ModuleFeatures == ModuleStatus::SYCL_ONLY' ?